### PR TITLE
k8s migration: add handler for multi proxy to forward

### PR DESCRIPTION
**Pull request by chrisohaver**
Each release now can have a  _postProcess_ function defined: (from code comments) 

> _postProcess_ is a post processing action to take on the corefile as a whole.  Used for complex migration  tasks that dont fit well into the modular plugin/option migration framework. For example, when the action on a plugin would need to extend beyond the scope of that plugin (affecting other plugins, or server blocks, etc). e.g. Splitting plugins out into separate server blocks.

Adds function _breakForwardStubDomainsIntoServerBlocks_() to handle splitting multiple forwards into separate blocks as a post processing step.

The multi-proxy->forward migration goes like this:
1. multiple proxy plugins exist in one server block.
2. existing migration turns each proxy into a forward.
3. now there are multiple forward plugins in one serverblock, which would be invalid if we left them like that.
4. the postProcess action then splits the multiple forwards out into separate blocks.

Strictly speaking, the process splits out any forwards to non-default zones  (".").  So if only one forward  exists, and its not for the default zone, then it would get split out into a separate server block even though it doesn't need to.  But this is OK (and a corner case if we ever see it).

### CODE REVIEW

File: kubernetes/migration/migrate_test.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package migration
 
 import (
+	"github.com/andreyvit/diff"
```
Comment (rajansandeep) [inline kubernetes/migration/migrate_test.go]: order of import.

File: kubernetes/migration/migrate_test.go [hunk 2]
```diff
@@ -145,7 +199,7 @@ func TestMigrate(t *testing.T) {
 			}
 
 			if result != testCase.expectedCorefile {
-				t.Errorf("expected %v; got %v", testCase.expectedCorefile, result)
+				t.Errorf("expected -> got diffs:\n%v", diff.LineDiff(testCase.expectedCorefile, result))
```
Comment (rajansandeep) [inline kubernetes/migration/migrate_test.go]: Is this used due to the order of the `stubDomain` in the expectedCorefile output changing?

File: kubernetes/migration/versions.go [hunk 3]
```diff
@@ -1056,3 +1066,43 @@ var proxyRemoveHttpsGoogleProtocol = func(o *corefile.Option) (*corefile.Option,
 	}
 	return o, nil
 }
+
+func breakForwardStubDomainsIntoServerBlocks(cf *corefile.Corefile) (*corefile.Corefile, error) {
```
Comment (rajansandeep) [inline kubernetes/migration/versions.go]: If I understand this correctly, the migration of `proxy` to `forward` takes place first and then this function breaks them up into separate stanzas. Neat!

File: kubernetes/migration/migrate_test.go (hunk 2 above)
Comment (chrisohaver) [reply kubernetes/migration/migrate_test.go]: No, the expected value was getting long, and I couldn't easily spot the difference. So I just use this tool to show which lines were different... turns out it was a white space difference (tab vs space), but knowing which lines it was helped.

**Pull request merged**